### PR TITLE
[8.0][stock_account] Changed the rounding at the moment to create the journal item from the move

### PR DESCRIPTION
**Pull request by josemoralesp**
There is difference between the account move line created from the invoice with the created from the stock move. This difference is because the rounding use at the moment to created the account move line from the move is different, to avoid this difference we are using the 'Account' rounding to create the account move line with the same amount that the journal created from the invoice

### CODE REVIEW

**Comment by nim-odoo** [comment]
Hello,

Thank you for the PR. However, that does not seem correct. First, I don't understand why use the 'Account' precision instead of the 'Product Price' precision. Then, if this precision is higher than the currency you will end up with such situation:
http://pix.toile-libre.org/upload/original/1435222806.png
I set up the accounting / product price precision to 4, while my currency precision is 2. We end up with journal entries which have an accuracy higher than the currency.

Or is there something I am missing?

**Comment by hbto** [comment]
Dear @nim-odoo 

thing here is that there should be consistency among the rounding when creating & writing account.move.line object, you _should__not_ have in one side journal entry lines been written with "Product Price" and the other side with "Account", if not we will endup with #7276

What is discussed here is not having 4 or 2 decimal for rounding, what is being discussed where is
accounting things shall bear accounting round.
other things shall bear their own rounding 

In other words, do not mix other roundings with accounting things, that is, all journal entry lines that are to be written within the system should only use "Account", regardless if end user decides to use 4 or 2 digits.

Best regards

**Comment by nim-odoo** [comment]
@hbto : thanks a lot for the clarification.

File: addons/stock_account/tests/test_reconciliation.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                           {'active_model': 'stock.picking',
+                            'active_id': pick_id,
+                            'active_ids': [pick_id]})
+        # Checking if the invoice was created
+        self.assertTrue(len(invoice_ids) > 0,
+                        'The invoice was not created')
+        # Validating the invoice
+        self.invoice.signal_workflow(cr, uid, invoice_ids, 'invoice_open')
+        for invo in invoice_ids:
+            invo_brw = self.invoice.browse(cr, uid, invo)
```
Comment (nim-odoo) [inline addons/stock_account/tests/test_reconciliation.py]: Why not:
```python
for invo in self.invoice.browse(cr, uid, invoice_ids):
```
?

File: addons/stock_account/tests/test_reconciliation.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            # Checking if the invoices were validated
+            self.assertTrue(invo_brw.state == 'open',
+                            'The invoice {invo} was not '
+                            'validated'.format(invo=invo_brw.name))
+        # Get the account move line created from the picking
+        p_move_ids = self.move_line.search(cr, uid,
```
Comment (nim-odoo) [inline addons/stock_account/tests/test_reconciliation.py]: Maybe check that p_move_ids is not empty?

File: addons/stock_account/tests/test_reconciliation.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        # Verifying if the picking was validated
+        self.assertTrue(pick_brw.state == 'done',
+                        'The picking was not transfered')
+        # Creating the invoice from the picking
+        value = self.stock_invoice.\
+            default_get(cr, uid, [],
+                        {'active_model': 'stock.picking',
+                         'active_id': pick_id,
+                         'active_ids': [pick_id]})
+        invo_id = self.stock_invoice.create(cr, uid, value,
```
Comment (nim-odoo) [inline addons/stock_account/tests/test_reconciliation.py]: Just being picky here, but I would use wiz_id instead of invo_id, as already done in elsewhere: https://github.com/odoo/odoo/blob/8.0/addons/stock_invoice_directly/test/stock_invoice_directly.yml#L44

**Comment by qdp-odoo** [comment]
i don't see what use case you're trying to fix... Do you have a real one to share?

what i can say, so far, is that you _have_ to use currency_round. That's the only way to ensure the value being written in the accounting is correct if your currency rounding is 0.05 for example. Your patch doesn't support that

**Comment by josemoralesp** [comment]
Hello @all

@nim-odoo I made  changes in the code to follow your recomendations.

@qdp-odoo The root of the problem is that you are using different rounding to create acounting movements. No matter who is creating the accounting movement the rounding used to generate it must be the same(Accounting), if not your are creating accounting inconsistencies.

@qdp-odoo You can see it in the example sent by @nim-odoo [1], the amount of the journal item created from the stock is 84.1267 and amount of the journal items created from the invoice is 84.13. The amount must be the same, if you try to reconcile that you will have a gap and that there should be no difference. For this reason I repet no matter what model is creating the accounting movements, the rounding used to create it must be the same(Accounting), to avoid gap where none exists.

All accounting movements must be created with the same rounding 

[1] http://pix.toile-libre.org/upload/original/1435222806.png

**Comment by nhomar** [comment]
@qdp-odoo 

IMHO to create any account move you must use accounting rounding, if you will make currency conversion then currency rounding.

What we are trying to fix here is that you can not make account moves with other than Accounting rounding.

We have a real database with the use case, but actually you are seeing the example that put @nim-odoo which he try to indicate they are the same but they are not.

Then if you try to compare both amoutns (which are the ones whcih extract amount for inventory valuation) you will face a difference which is unreciliable (using reconcile name as the preocedural proceess not the technical process in odoo).

With such difference the accountants face this as inconsistent.

Regards.
